src/utils.ts

```diff
@@ -109,6 +109,18 @@ export function checkAInB(a: BBox, b: BBox): boolean {
   return insideX && insideY;
 }
 
+export function clonePointsArray(points: SATVector[]): SATVector[] {
+  let pointsClone = [];
+
+  for (var i = 0; i < points.length; i++) {
+    let point = points[i];
+
+    pointsClone.push(new SATVector(point.x, point.y));
+  }
+
+  return pointsClone;
+}
```

Comment:
```suggestion
  return Array.from(points, (_: unknown, index: number) =>
    points[index].clone()
  );
```

this would do the same but simpler